File: ExpenseManagement/app/src/main/java/com/example/expensemanagement/MainActivity.java
package com.example.expensemanagement;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.view.View.OnClickListener;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

import static android.provider.AlarmClock.EXTRA_MESSAGE;

public class MainActivity extends AppCompatActivity {
    private EditText mEmail, mPass;
    private Button btnLogin;
    private TextView mForgetPassword;
    private TextView mSignupHere;

    private FirebaseAuth mAuth;
    private ProgressDialog mDialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mAuth =FirebaseAuth.getInstance();
        if(mAuth.getCurrentUser()!=null){
            startActivity(new Intent(getApplicationContext(),HomeActivity.class));

        }
        mDialog= new ProgressDialog(this);

        loginDetails();
    }

    private void loginDetails(){
        mEmail = findViewById(R.id.email_login);
        mPass = findViewById(R.id.password_login);
        btnLogin = findViewById(R.id.btn_login);
        mForgetPassword = findViewById(R.id.forget_password);
        mSignupHere = findViewById(R.id.signup_reg);

        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String email = mEmail.getText().toString().trim();
                String pass = mPass.getText().toString().trim();

                if(TextUtils.isEmpty(email)){
                    mEmail.setError("Email require...");
                    return;
                }
                if(TextUtils.isEmpty(pass)){
                    mPass.setError("Password require...");
                    return;
                }

                mDialog.setMessage("Processing...");
                mDialog.show();
                mAuth.signInWithEmailAndPassword(email,pass).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if(task.isSuccessful()){
                            mDialog.dismiss();
                            Intent intent = new Intent(MainActivity.this,HomeActivity.class);
                            startActivities(new Intent[]{intent});
                            Toast.makeText(getApplicationContext(),"Login Successful...",Toast.LENGTH_SHORT).show();
                        }
                        else {
                            mDialog.dismiss();
                            Toast.makeText(getApplicationContext(),"Login Field...",Toast.LENGTH_SHORT).show();

                        }
                    }
                });

            }
        });


        //dang ky
        mSignupHere.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
       //         startActivities(new Intent[getApplicationContext(),RegistrationActivity.class]);

                Intent intent  =   new Intent(MainActivity.this, RegistrationActivity.class);
                startActivity(intent);

            }
        });

        mForgetPassword.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent  =   new Intent(MainActivity.this, RegistrationActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Ungdungbetoanhoc/app/src/main/java/com/example/ungdungbetoanhoc/MainActivity.kt
package com.example.ungdungbetoanhoc

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        var linearLayout =findViewById<LinearLayout>(R.id.linearlayout)

        var buttonCong= findViewById<Button>(R.id.buttoncong)
        var buttonTru= findViewById<Button>(R.id.buttontru)
        var buttonNhan= findViewById<Button>(R.id.buttonnhan)
        var buttonChia= findViewById<Button>(R.id.buttonchia)
        var edittextThu1=findViewById<EditText>(R.id.edittextthu1)
        var edittextThu2=findViewById<EditText>(R.id.edittextthu2)
        var textviewKetqua=findViewById<TextView>(R.id.textviewketqua)
        buttonCong.setOnClickListener {
            var sothu1: Int= (edittextThu1.text.toString()).toInt()
            var sothu2: Int= (edittextThu2.text.toString()).toInt()
            var ketqua =sothu1+sothu2
            textviewKetqua.text =ketqua.toString()
        }
        buttonTru.setOnClickListener {
            var sothu1: Int= (edittextThu1.text.toString()).toInt()
            var sothu2: Int= (edittextThu2.text.toString()).toInt()
            var ketqua =sothu1 - sothu2
            textviewKetqua.text =ketqua.toString()
        }
        buttonNhan.setOnClickListener {
            var sothu1: Int= (edittextThu1.text.toString()).toInt()
            var sothu2: Int= (edittextThu2.text.toString()).toInt()
            var ketqua =sothu1 * sothu2
            textviewKetqua.text =ketqua.toString()
        }
        buttonChia.setOnClickListener {
            var sothu1: Int= (edittextThu1.text.toString()).toInt()
            var sothu2: Int= (edittextThu2.text.toString()).toInt()
            var ketqua =sothu1 / sothu2
            textviewKetqua.text =ketqua.toString()
        }




    }
}